### DISCUSSION TITLE
More diagnostic fixes

### DISCUSSION THREAD

**Discussion by sam-mccall** [clangd/clangd#954] [Ideas]
Clang mostly only emits those that are nice to read, which usually means short.

But some diagnostics have really obvious/useful fixes, e.g.:
 - unhandled enum cases
 - member initializers in the wrong order

Unclear if the logic should go in sema, clangd, or even clang-tidy.

**Comment by kadircet** [comment]
I believe this is something we value in general and try to improve as we stumble upon such cases. but the issue is too vague on its own to be acted directly. Hence closing this one.

But I am creating a more general "roadmap" document, in which I put an item around this one. Hopefully I'll make it available soon.

**Comment by kadircet** [comment]
https://github.com/clangd/clangd/issues/77 is a good example of this.
